beets: 2.2.0 -> 2.3.0

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
beets: 2.2.0 -> 2.3.0

https://github.com/beetbox/beets/releases/tag/v2.3.0

doronbehar committed:
beets-unstable: 2.2.0-unstable-2025-03-12 -> 2.3.0

doronbehar commented:
Darwin failures are due to gst-python 1.26.0 failing to build:

- https://github.com/NixOS/nixpkgs/issues/405167

antonmosich commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 405156`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 17 packages built:</summary>
  <ul>
    <li>beets (beetsPackages.beets, beetsPackages.beets-stable)</li>
    <li>beets-unstable (beetsPackages.beets-unstable)</li>
    <li>beets-unstable.dist (beetsPackages.beets-unstable.dist)</li>
    <li>beets-unstable.doc (beetsPackages.beets-unstable.doc)</li>
    <li>beets-unstable.man (beetsPackages.beets-unstable.man)</li>
    <li>beets.dist (beetsPackages.beets-stable.dist, beetsPackages.beets.dist)</li>
    <li>beets.doc (beetsPackages.beets-stable.doc, beetsPackages.beets.doc)</li>
    <li>beets.man (beetsPackages.beets-stable.man, beetsPackages.beets.man)</li>
    <li>beetsPackages.audible</li>
    <li>beetsPackages.audible.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
    <li>supergee</li>
  </ul>
</details>

doronbehar commented:
Thanks for the report, and thanks for opening the upstream issue! Let's give them a week or so. In the meantime, I'll mark this as a draft. For reference:

- https://github.com/geigerzaehler/beets-alternatives/issues/136

doronbehar committed:
beetsPackages.alternatives: 0.13.2 -> 0.13.3

Diff: https://github.com/geigerzaehler/beets-alternatives/compare/refs/tags/v0.13.2...refs/tags/v0.13.3

Changelog: https://github.com/geigerzaehler/beets-alternatives/blob/v0.13.3/CHANGELOG.md

doronbehar committed:
beetsPackages.audible: Put beets in nativeBuildInputs

doronbehar committed:
beetsPackages.copyartifacts: Put beets in nativeBuildInputs

Has no real difference on the results, but it makes a bit more sense to
not remove beets from the setup.py file, and depend on it only via
nativeBuildInputs. This is also consistent with other plugins, see:

https://github.com/NixOS/nixpkgs/pull/257943#discussion_r2083635799

doronbehar committed:
beetsPackages.copyartifacts: Small consistency empty lines

dansbandit committed:
beetsPackages.filetote: init at 1.0.1

Co-Authored-By: Doron Behar <doron.behar@gmail.com>

doronbehar commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 405156`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 21 packages built:</summary>
  <ul>
    <li>beets (beetsPackages.beets, beetsPackages.beets-stable)</li>
    <li>beets-unstable (beetsPackages.beets-unstable)</li>
    <li>beets-unstable.dist (beetsPackages.beets-unstable.dist)</li>
    <li>beets-unstable.doc (beetsPackages.beets-unstable.doc)</li>
    <li>beets-unstable.man (beetsPackages.beets-unstable.man)</li>
    <li>beets.dist (beetsPackages.beets-stable.dist, beetsPackages.beets.dist)</li>
    <li>beets.doc (beetsPackages.beets-stable.doc, beetsPackages.beets.doc)</li>
    <li>beets.man (beetsPackages.beets-stable.man, beetsPackages.beets.man)</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.audible</li>
    <li>beetsPackages.audible.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
    <li>beetsPackages.filetote</li>
    <li>beetsPackages.filetote.dist</li>
    <li>supergee</li>
  </ul>
</details>

antonmosich reviewed (approved):
LGTM

doronbehar commented:
I'll give @jwillikers and the other maintainers a day or two to respond, since the changes might be considered significant, and I'll merge it afterwards.

aszlig reviewed (approved):
Without actually testing the changes, LGTM. Thanks :-)
